### PR TITLE
Segregate tests + Script to upgrade components

### PR DESCRIPTION
**Pull request by iamrk04**
This PR has the following changes:
- add script to automatically upgrade the required parts of components before kicking off release.
- segregate tests for all aml-benchmark components to parallelly run them, this will allow us to run all the tests in less time.
- improve running command in subprocess to raise the error in a better way, this eases the error message propagation and debugging process.

### CODE REVIEW

File: assets/aml-benchmark/scripts/_internal/upgrade_components.py [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+            file.write(spec_str)
+    except Exception as e:
+        is_error = True
+        error = str(e)
+    return is_error, error, component_path, name
+
+
+def main():
+    """Entry function."""
+    components = _get_all_components_spec()
```
Comment (arun-rajora) [inline assets/aml-benchmark/scripts/_internal/upgrade_components.py:164]: Can we create flags to make upgrade environment, ignore resource_manager configurable? If possible, let us specify the environment version to pin or component version to pin to (instead of latest). Thanks!

Comment (iamrk04) [reply assets/aml-benchmark/scripts/_internal/upgrade_components.py:195]: Added argument env_version` to control env upgrade (more details in readme.md). All the versions are now pinned to a specific version instead of `latest` label. Ignoring components for release is something that we should control within the script, there is a global variable for it which can be changed. Not adding this as an argument at the moment, does not seem to be worth doing at the moment.

File: assets/aml-benchmark/components/batch-benchmark-inference/spec.yaml [hunk 2]
```diff
@@ -4,7 +4,7 @@ type: pipeline
 name: batch_benchmark_inference
 display_name: Batch Benchmark Inference
 description: Components for batch endpoint inference
-version: 0.0.5
+version: 0.0.6
```
Comment (arun-rajora) [inline assets/aml-benchmark/components/batch-benchmark-inference/spec.yaml:7]: Can we undo these changes in the spec? Thanks!

**Review by arun-rajora — DISMISSED** [review]

**Review by arun-rajora — APPROVED** [review]
